typescript/src/schema/resource-database.ts

```diff
@@ -185,6 +185,32 @@ export class ResourceDatabase {
     return params;
   }
 
+  private getName(pattern: string): string {
+    const elements = pattern.split('/');
+    const name = [];
+    // Multi pattern like: `projects/{project}/cmekSettings`, we need to append `cmekSettings` to the name.
+    // Or it will be duplicate with `project/{project}`
+    // Iterate the elements, if it comes in pairs: user/{userId}, we take `userId` as part of the name.
+    // if it comes as `profile` with no following `/{profile_id}`, we take `profile` as part of the name.
+    // So for pattern: `user/{user_id}/profile/blurbs/{blurb_id}`, name will be `userId_profile_blurbId`
+    while (elements.length > 0) {
```

Comment:
Overall implementation LGTM, thank you!

Reply:
fixed the problem, and add `{organization=**}` case to the unit test. Thank you!

Reply:
Add one more special use case to the uni test: `{unknown_path=**}/tasks/{task}/result`. Thanks for the script of checking all odd number pairs of the resource pattern :)